.github/workflows/ci.yml

```diff
@@ -0,0 +1,39 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+jobs:
+  spm:
+    name: Build and Test on Xcode 16
+    runs-on: macos-15
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Select Xcode Version
+        run: sudo xcode-select --switch /Applications/Xcode_16.app/Contents/Developer
+      - name: Build and Test Framework
+        run: |
+          xcrun swift test -c release --enable-code-coverage -Xswiftc -enable-testing
+
+  linux:
+    name: Build and Test on Linux
+    runs-on: ubuntu-latest
+    container: swift:6.0
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Build and Test Framework
+        run: swift test -c release --enable-code-coverage -Xswiftc -enable-testing
```

Comment:
this is becoming a habit of mine in open source libs: test in release!

makes things a bit slower in CI but I have caught a few esoteric issues this way.

Reply:
though testing in release on macOS yields an error executing the CLI, which... I don't get why. works on linux at least.

Reply:
Oh interesting! I've never thought about testing in release. Let's try it out!